command/volume_create_host.go

```diff
@@ -0,0 +1,228 @@
+	vol, err := decodeHostVolume(ast)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error decoding the volume definition: %s", err))
+		return 1
+	}
+
+	req := &api.HostVolumeCreateRequest{
+		Volumes: []*api.HostVolume{vol},
+	}
+	vols, _, err := client.HostVolumes().Create(req, nil)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error creating volume: %s", err))
+		return 1
+	}
+	for _, vol := range vols {
+		// note: the command only ever returns 1 volume from the API
+		c.Ui.Output(fmt.Sprintf(
+			"Created host volume %s with ID %s", vol.Name, vol.ID))
+	}
+
+	// TODO(1.10.0): monitor so we can report when the node has fingerprinted
+
+	return 0
+}
+
+func decodeHostVolume(input *ast.File) (*api.HostVolume, error) {
+	var err error
+	vol := &api.HostVolume{}
+
+	list, ok := input.Node.(*ast.ObjectList)
+	if !ok {
+		return nil, fmt.Errorf("error parsing: root should be an object")
+	}
+
+	// Decode the full thing into a map[string]interface for ease
+	var m map[string]any
+	err = hcl.DecodeObject(&m, list)
+	if err != nil {
+		return nil, err
+	}
+
+	// Need to manually parse these fields/blocks
+	delete(m, "capability")
+	delete(m, "constraint")
+	delete(m, "capacity_max")
+	delete(m, "capacity_min")
+	delete(m, "type")
+
+	// Decode the rest
+	err = mapstructure.WeakDecode(m, vol)
+	if err != nil {
+		return nil, err
+	}
+
+	capacityMin, err := parseCapacityBytes(list.Filter("capacity_min"))
+	if err != nil {
+		return nil, fmt.Errorf("invalid capacity_min: %v", err)
+	}
+	vol.RequestedCapacityMin = capacityMin
+	capacityMax, err := parseCapacityBytes(list.Filter("capacity_max"))
+	if err != nil {
+		return nil, fmt.Errorf("invalid capacity_max: %v", err)
+	}
+	vol.RequestedCapacityMax = capacityMax
+
+	if o := list.Filter("constraint"); len(o.Items) > 0 {
+		if err := parseConstraints(&vol.Constraints, o); err != nil {
+			return nil, fmt.Errorf("invalid constraint: %v", err)
+		}
+	}
+	if o := list.Filter("capability"); len(o.Items) > 0 {
+		if err := parseHostVolumeCapabilities(&vol.RequestedCapabilities, o); err != nil {
+			return nil, fmt.Errorf("invalid capability: %v", err)
+		}
+	}
+
+	return vol, nil
+}
+
+func parseHostVolumeCapabilities(result *[]*api.HostVolumeCapability, list *ast.ObjectList) error {
+	for _, o := range list.Elem().Items {
+		valid := []string{"access_mode", "attachment_mode"}
+		if err := helper.CheckHCLKeys(o.Val, valid); err != nil {
+			return err
+		}
+
+		ot, ok := o.Val.(*ast.ObjectType)
+		if !ok {
+			break
+		}
+
+		var m map[string]any
+		if err := hcl.DecodeObject(&m, ot.List); err != nil {
+			return err
+		}
+		var cap *api.HostVolumeCapability
+		if err := mapstructure.WeakDecode(&m, &cap); err != nil {
+			return err
+		}
+
+		*result = append(*result, cap)
+	}
+
+	return nil
+}
+
+func parseConstraints(result *[]*api.Constraint, list *ast.ObjectList) error {
+	for _, o := range list.Elem().Items {
+		valid := []string{
+			"attribute",
+			"distinct_hosts",
```

Comment:
It was! We ripped it out with the rest of HCL1 jobspec parsing. Because everything _other_ than jobspecs is still HCL1, I needed to resurrect this from the code we removed.